docs/manual/constant-value-checker.tex

```diff
@@ -334,6 +334,30 @@
 that possibility, or verify the lack of overflow using some other tool or
 manual analysis.
 
+\sectionAndLabel{Strings can be null in concatenations}{null-strings-concats}
+
+The Constant Value Checker considers all elements of a String concatenation
+non-null by default. The \code{-AnullStringsConcatenation} command-line option
+makes the checker more sound by analyzing the possibility of each element to
+be null. Only String literals, primitive types and effectively final variables
+are considered non-null. Figure~\ref{fig-null-strings-concat} gives an example
+of behavior with and without the \code{-AnullStringsConcatenation} option.
+
+\begin{figure}
+\begin{Verbatim}
+  ...
+  // This String can be null
+  @StringVal("a") String string1;
+  // The possible value "nullb" is omitted
+  @StringVal("ab") String string2 = string1 + "b";
+\end{Verbatim}
+\caption{With the \code{-AnullStringsConcatenation} command-line option, the
+Constant Value Checker considers that the first String can be null.}
+\label{fig-null-strings-concat}
+\end{figure}
+
+It is recommended to use the Nullness Checker before using the Constant Value Checker
+with this command-line option enabled for a better analysis on nullable elements.
```

Comment:
This recommendation doesn't make sense, because using the nullness checker doesn't help. I would suggest either:
* use the nullness checker, and do NOT use this option, or
* use this option

Reply:
i removed this altogether. Hopefully this is an improvement.